uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointDocs.java

```diff
@@ -705,9 +705,12 @@ void createOidcIdentityProvider() throws Exception {
                 fieldWithPath("config.setForwardHeader").optional(false).type(BOOLEAN).description("Only effective if Password Grant enabled. Set X-Forward-For header in Password Grant request to this identity provider."),
                 fieldWithPath("config.jwtClientAuthentication").optional(null).type(OBJECT).description("<small><mark>UAA 76.5.0</mark></small> Only effective if relyingPartySecret is not set or null. Creates private_key_jwt client authentication according to OIDC or OAuth2 (RFC 7523) standard. "+
                         "<br>Please note that you can precise the created JWT for client authentication, e.g. if your IdP follows OAuth2 standard according to RFC 7523. For standard OIDC compliance, set true without any further sub-parameters. The supported sub-parameters are " +
-                        "<ul><li>    `kid` <small><mark>UAA 76.18.0</mark></small> Custom key from your defined keys, defaults to `activeKeyId` </li>" +
-                        "<li>        `iss` Custom issuer, see RFC 7523, defaults to `relyingPartyId` </li>" +
-                        "<li>        `aud` Custom audience, see RFC 7523, defaults to `tokenUrl` </li></ul>"),
+                        "<ul><li>    `kid`  <small><mark>UAA 76.18.0</mark></small> Custom key from your defined keys, defaults to `activeKeyId` </li>" +
+                        "<li>        `key`  <small><mark>UAA 77.3.0</mark></small> Optional custom private key, used to generate the client JWT signature </li>" +
```

Comment:
client_secret_jwt is not wanted from SAP side because with this you have a symmentric/shared key - again. This is same level of security like a secret .
The private_key_jwt is an asymmetric mechanism, see https://datatracker.ietf.org/doc/draft-ietf-oauth-v2-1/ , section 2.4 ->

It is RECOMMENDED to use asymmetric (public-key based) methods for
   client authentication such as mTLS [RFC8705] or using signed JWTs
   ("Private Key JWT") in accordance with [RFC7521] and [RFC7523] (in
   [OpenID] defined as the client authentication method
   private_key_jwt).  When such methods for client authentication are
   used, authorization servers do not need to store sensitive symmetric
   keys, making these methods more robust against a number of attacks.

Reply:
I'm referring to the doc.

In the doc I linked form you, they list each item as "OPTIONAL" or "REQUIRED". This is a clean way to identify configuration items. The docs here on 708-713 say Optional on some and each has a default telling me they are all optional.

Reply:
ok will update documentation

Reply:
please see now doc below